Formula/c/cucumber-cpp.rb

```diff
@@ -61,24 +63,41 @@ def install
       THEN("^A then statement$") {
       }
     EOS
-    system ENV.cxx, "test.cpp", "-o", "test", "-I#{include}", "-L#{lib}",
-           "-lcucumber-cpp", "-I#{boost.opt_include}",
-           "-L#{boost.opt_lib}", "-lboost_regex", "-lboost_system",
-           "-lboost_program_options", "-lboost_filesystem", "-lboost_chrono",
-           "-pthread"
+
+    cxx_args = %W[
+      -std=c++17
```

Comment:
Happy to adopt a more idiomatic way if there is one, just was finding that this `system` call was getting long and unwieldy

Reply:
can we get it from `pkg-config`?

Reply:
I've gotten surprisingly far in this world without ever really understanding how pkg-config works 😅  –  I'll take a look at some other formulas to educate myself on how it might apply here.

Reply:
@SMillerDev When I first posted this there had been a much longer args list (due to boost), now not so much. Wondering if you could elaborate on how `pkg-config` would help here, was it to avoid needing to specify other lib paths like [here](https://github.com/Homebrew/homebrew-core/pull/160521/files#diff-993459b6da143ab8ee11d4ac45a5f8ce1fe593712947cb5de865f9dfce8f4fb7R75-R78)?

Reply:
Take a look here: https://github.com/Homebrew/homebrew-core/blob/49ff10e5afb75dc3f41b1c884bf6f5146e961926/Formula/a/at-spi2-core.rb#L70-L71

Reply:
Ah, ok thank you. I'm not sure that in this case it's really a readability improvement, but do you think it's preferable from a functional perspective? Or it is better to just keep it simple, and avoid adding `pkg-config` as a dep here, where there isn't some bigger need for it? For example cucumber-cpp's own build process doesn't depend on it.

Reply:
Since `pkg-config` isn't anywhere in the dep tree, I don't think it's necessary. But I'll let @SMillerDev make the final call on that.

Reply:
I like it when packages ship a pkg-config file, since it tests the file then. But if it doesn't there's no benefit.